carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/services/IntegrationEvent.kt

```diff
@@ -3,16 +3,23 @@ package dk.cachet.carp.common.application.services
 import dk.cachet.carp.common.application.Immutable
 import dk.cachet.carp.common.application.ImplementAsDataClass
 import kotlinx.serialization.Polymorphic
-import kotlinx.serialization.Serializable
 
 
 /**
  * An event raised by an application service [TApplicationService].
  *
  * Integration events need to be immutable.
  */
-@Serializable
 @Polymorphic
 @Immutable
 @ImplementAsDataClass
-abstract class IntegrationEvent<out TApplicationService : ApplicationService<out TApplicationService, *>>
+interface IntegrationEvent<out TApplicationService : ApplicationService<out TApplicationService, *>>
+{
+    /**
+     * All events related to the same aggregate ID are handled in order.
+     *
+     * In case the event pertains to an aggregate root,
+     * specify its ID to ensure correct handling of business logic in the domain.
+     */
+    val aggregateId: String?
```

Comment:
Not sure what to make of the nullable `aggregateId` - If for some reason it is `null` when an event is passed to the event bus, does it mean that ordering is not important for this event and any (fallback) key can be used _or_ should it be treated exceptionally and result in an error?

Reply:
Well-spotted!

The intent was to convey ordering is not important, but I see that is not properly documented. Neither would treating `null` as an `aggregateId` be desirable: we don't want to guarantee ordering for all integration events which pass `null` here.

I think I'm happy with passing `null` here for the API, unless you can come up with a better suggestion. So my suggestion is to simply make this clear in the documentation.

Reply:
I'm onboard with clearing up the documentation, since event constructors ensure non-null ids where applicable